test/cpp/api/any.cpp

```diff
@@ -0,0 +1,200 @@
+#include <catch.hpp>
+
+#include <torch/torch.h>
+
+#include <torch/detail/any.h>
+#include <torch/nn/modules/any.h>
+
+#include <algorithm>
+#include <string>
+
+using namespace torch;
+using namespace torch::nn;
+using namespace torch::detail;
+
+using Catch::StartsWith;
+
+TEST_CASE("any") {
+  SECTION("gets the correct value for the right type") {
+    SECTION("int") {
+      auto any = make_any(5);
+      // const and non-const types have the same typeid()
+      REQUIRE(any.try_get<int>() != nullptr);
```

Comment:
I guess, if you wanted to test that it actually returned 5, you'd have to say `*any.try_get<int>() == 5`, which would then promptly segfault if you had got it wrong?